controllers/pipeline/pipeline_adapter_test.go

```diff
@@ -229,6 +231,18 @@ var _ = Describe("Pipeline Adapter", Ordered, func() {
 		Expect(compositeSnapshot == nil).To(BeTrue())
 	})
 
+	It("ensures pipelines as code labels and annotations are propagated to the snapshot", func() {
+		snapshot, err := adapter.prepareSnapshotForPipelineRun(testpipelineRunBuild, hasComp, hasApp)
+		Expect(err).To(BeNil())
+		Expect(snapshot).ToNot(BeNil())
+		annotation, found := snapshot.GetAnnotations()["pac.test.appstudio.openshift.io/on-target-branch"]
+		Expect(found).To(BeTrue())
+		Expect(annotation).To(Equal("[main,master]"))
+		label, found := snapshot.GetLabels()["pac.test.appstudio.openshift.io/event-type"]
+		Expect(found).To(BeTrue())
+		Expect(label).To(Equal("pull_request"))
+	})
+
```

Comment:
You might consider adding tests for adverse scenarios, for example ensuring keys without the matching prefix are **NOT** copied to the destination maps. As far as I can tell this test provides full code coverage.
You can look it over with `go tool cover -html cover.out -o cover.html` and then reviewing the report in your browser.

(optional)

Reply:
A few more tests have been added. Please take a look.

Reply:
:+1: